helm-charts/splunk-otel-collector/templates/daemonset.yaml

```
@@ -138,6 +138,47 @@ spec:
               mountPath: /var/log
             - name: varlibdockercontainers
               mountPath: /var/lib/docker/containers
+        {{- if $agent.securityContext }}
+        {{- if not (eq (toString $agent.securityContext.runAsUser) "0" ) }}
```

I think we should enable it if both runAsUser and runAsGroup are provided
```suggestion
        {{- if and $agent.securityContext.runAsUser $agent.securityContext.runAsGroup }}
```

To make it work with one if statement I'd recommend bringing back `securityContext: {}` by default instead of removing it

---

applied